mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java

```diff
@@ -2329,4 +2331,8 @@ public int scaleStage(MantisStageMetadataImpl stageMetaData, int numWorkers, Str
             return newNumWorkerCount;
         }
     }
+
+    private String getResourceCluster() {
+        return mantisJobMetaData.getJobDefinition().getResourceCluster().map(ClusterID::getResourceID).orElse("mesos");
+    }
```

Comment:
I prefer we restrict the creation of ClusterIDs to a few places - ideally, one location.